core/src/replay_stage.rs

```diff
@@ -493,14 +493,24 @@ impl ReplayStage {
                 !is_locked_out
             })
             .map(|bank| {
-                (
-                    bank,
-                    tower.collect_vote_lockouts(
-                        bank.slot(),
-                        bank.vote_accounts().into_iter(),
-                        &ancestors,
-                    ),
-                )
+                let (lockouts, total_epoch_stakes) = tower.collect_vote_lockouts(
+                    bank.slot(),
+                    bank.vote_accounts().into_iter(),
+                    &ancestors,
+                );
+                let mut w_bank_forks = bank_forks.write().unwrap();
+                for (fork, stake_lockout) in lockouts.iter() {
+                    w_bank_forks.cache_fork_confidence(
+                        *fork,
+                        Confidence::new(
```

Comment:
@aeyakovenko, @CriesofCarrots, these aren't really "fork" metadata, they're morel like "bank" specific metadata. These stake_lockout.stake() values and stake_lockout.lockout() values will never change for a particular bank b/c they are based on the vote state observed in this bank. Thus, even if for instance, validators switch to a different fork, or if they continue to vote on this same fork, none of the values will update. 

Is the plan to track which banks are part of which forks so that you can update fork information as new lockout/stake information is calculated for banks at the tip of each fork?

Reply:
My understanding is that, because we are using the fork ids from StakeLockout as the cache key to update (as opposed to the current `bank.slot()`), and because of `update_ancestor_lockouts` in `Tower::collect_vote_lockouts()`, the values in the cache will change as subsequent banks are processed.
(But I'll admit, I found it tough to grok how replay_stage does this, so please correct me if I'm wrong!)

What I don't think will update is the stake-weighted lockouts that I just added. Wondering if that cache process needs some modification.

Reply:
@carllin @CriesofCarrots a banks “finality”, is an aggregate of all
the children.

Reply:
Yes, exactly. I've fixed up the stake-weighted lockouts cache to aggregate properly.
I'll work on writing an integration test that demonstrates how the cached values increase/accumulate.